include/zephyr/pixel/image.h

```diff
@@ -0,0 +1,203 @@
+ *
+ * @param img Image being processed.
+ * @param vbuf Video buffer that receives the image data.
+ * @return 0 on success
+ */
+int pixel_image_to_vbuf(struct pixel_image *img, struct video_buffer *vbuf);
+
+/**
+ * @brief Convert an image to a new pixel format.
+ *
+ * An operation is added to convert the image to a new pixel format.
+ * If the operation to convert the image from the current format to a new format does not exist,
+ * then the error flag is set, which can be accessed as @c img->err.
+ *
+ * In some cases, converting between two formats requires an intermediate conversion to RGB24.
+ *
+ * @param img Image to convert.
+ * @param new_format A four-character-code (FOURCC) as defined by @c <zephyr/drivers/video.h>.
+ */
+int pixel_image_convert(struct pixel_image *img, uint32_t new_format);
+
+/**
+ * @brief Convert an image from a bayer array format to RGB24.
+ *
+ * An operation is added to convert the image to RGB24 using the specified window size, such
+ * as 2x2 or 3x3.
+ *
+ * @note It is also possible to use @ref pixel_image_convert to convert from bayer to RGB24 but
+ *       this does not allow to select the window size.
+ *
+ * @param img Image to convert.
+ * @param window_size The window size for convnerting, usually 2 (faster) or 3 (higher quality).
+ */
+int pixel_image_debayer(struct pixel_image *img, uint32_t window_size);
+
+/**
+ * @brief Resize an image.
+ *
+ * An operation is added to change the image size. The aspect ratio is not preserved and the output
+ * image size is exactly the same as requested.
+ *
+ * @param img Image to convert.
+ * @param width The new width in pixels.
+ * @param height The new height in pixels.
+ */
+int pixel_image_resize(struct pixel_image *img, uint16_t width, uint16_t height);
+
+/**
+ * @brief Apply a kernel operation on an image.
+ *
+ * Kernel operations are working on small blocks of typically 3x3 or 5x5 pixels, repeated over the
+ * entire image to apply a desired effect on an image.
+ *
+ * @param img Image to convert.
+ * @param kernel_type The type of kernel to apply as defined in <zephyr/pixel/kernel.h>
+ * @param kernel_size The size of the kernel operaiton, usually 3 or 5.
+ */
+int pixel_image_kernel(struct pixel_image *img, uint32_t kernel_type, int kernel_size);
+
+/**
+ * @brief Print an image using higher quality TRUECOLOR terminal escape codes.
+ *
+ * @param img Image to print.
+ */
+void pixel_image_print_truecolor(struct pixel_image *img);
+
+/**
+ * @brief Print an image using higher speed 256COLOR terminal escape codes.
+ *
+ * @param img Image to print.
+ */
+void pixel_image_print_256color(struct pixel_image *img);
+
+/** @cond INTERNAL_HIDDEN */
+
+/**
+ * @brief Add a operation processing step to an image.
+ *
+ * @note This is a low-level function only needed to implement new operations.
+ *
+ * The operation step will not be processed immediately, but rather added to a linked list of
+ * operations that will be performed at runtime.
```

Comment:
It will be added as latest operation, then maybe it should be clearer, naming it `pixel_image_append_operation()`?

Reply:
Absolutely! :pray: